File: app/src/main/java/com/divercity/app/features/profile/selectethnicity/SelectEthnicityActivity.java
package com.divercity.app.features.profile.selectethnicity;

import android.content.Context;
import android.content.Intent;

import com.divercity.app.core.base.BaseFragment;
import com.divercity.app.core.base.BaseFragmentActivity;

/**
 * Created by lucas on 17/10/2018.
 */

public class SelectEthnicityActivity extends BaseFragmentActivity {

    public static Intent getCallingIntent(Context context){
        return new Intent(context, SelectEthnicityActivity.class);
    }

    @Override
    protected BaseFragment fragment() {
        return SelectEthnicityFragment.newInstance();
    }

}

File: app/src/main/java/com/divercity/app/features/onboarding/selectethnicity/SelectEthnicityActivity.kt
package com.divercity.app.features.onboarding.selectethnicity

import android.content.Context
import android.content.Intent
import com.divercity.app.core.base.BaseActivity
import com.divercity.app.core.base.BaseFragment

/**
 * Created by lucas on 17/10/2018.
 */

class SelectEthnicityActivity : BaseActivity() {

    companion object {
        private const val INTENT_EXTRA_PARAM_PROGRESS = "progress"

        fun getCallingIntent(context: Context, progress : Int) : Intent {
            val intent = Intent(context, SelectEthnicityActivity::class.java)
            intent.putExtra(INTENT_EXTRA_PARAM_PROGRESS,progress)
            return intent
        }
    }

    override fun fragment(): BaseFragment = SelectEthnicityFragment.newInstance(
            intent.getIntExtra(INTENT_EXTRA_PARAM_PROGRESS, 0))
}
